File: 7KYU/Collatz Conjecture Length/solution.ts
export function collatz(n:number):number{
    var c = 1;  
     while(n != 1){
       c++;
       if(n % 2 == 0)
         n /= 2;
       else
         n = n * 3 + 1;
     }
     return c;
  }

File: 7KYU/Collatz Conjecture Length/solution.java
public class Collatz {
    public static long conjecture(long x) {
      int c = 1;
      
      while(x != 1) {
        c++;
        if(x % 2 == 0)
          x /= 2;
        else
          x = x * 3 + 1;
      }
      return c;
    }
}